http_service/tests/conftest.py

```diff
@@ -230,4 +232,20 @@ def mock_repo(tmpdir, monkeypatch):
         repo.add([str(test_file).encode("utf-8")])
         repo.commit(f"Base history {i}", user="bugbug")
 
+    # Save initial state for later cleanup
+    initial = repo.identify(id=True).strip()
+
+    # Add a new file when pulling from remote repo
+    def _clean(*args):
+        # Revert to initial state
+        repo.update(initial, clean=True)
+
+        # Then create a new "remote" file to emulate a pull
+        remote = repo_dir / "remote.txt"
+        remote.write_text("New remote file !", encoding="utf-8")
+        repo.add([str(remote).encode("utf-8")])
+        repo.commit("Pulled from remote", user="bugbug")
+
+    monkeypatch.setattr(bugbug.repository, "clean", _clean)
```

Comment:
What if we didn't monkeypatch it? We could have a local and remote like the tests in tests/test_repository.py.

Reply:
Done